test/utils/create_resources.go

```diff
@@ -59,7 +61,14 @@ func CreatePodWithRetries(c clientset.Interface, namespace string, obj *v1.Pod)
 	}
 	createFunc := func() (bool, error) {
 		_, err := c.CoreV1().Pods(namespace).Create(context.TODO(), obj, metav1.CreateOptions{})
-		if err == nil || apierrors.IsAlreadyExists(err) {
+		if err == nil {
+			return true, nil
+		}
+		if apierrors.IsAlreadyExists(err) {
+			if strings.Contains(err.Error(), errorGenerateNameConflict) {
```

Comment:
One nit: checking for the error text is fragile. Can you instead check whether `obj.Name` is empty?

Reply:
Sure. Fixed.